src/redux/books.js/booksSlice.js

```diff
@@ -0,0 +1,44 @@
+import { createSlice } from '@reduxjs/toolkit';
+
+const initialState = {
+  bookItem: [
+    {
+      id: 'item1',
+      title: 'Introduction to cybersecurity',
+      author: 'john Dow',
+      category: 'Fiction',
+    },
+    {
+      id: 'item2',
+      title: 'Secrete of success',
+      author: 'Napolioon Hill',
+      category: 'Nonfiction',
+    },
+    {
+      id: 'item3',
+      title: 'art of war',
+      author: 'muh Ndlovu',
+      category: 'Nonfiction',
+    },
+  ],
+};
+const booksSlice = createSlice({
+  name: 'book',
+  initialState,
+  reducers: {
+    addBook: (state, action) => {
+      state.bookItem.push(action.payload);
+    },
+    removeBook: (state, action) => {
+      const bookId = action.payload;
+      return {
+        ...state,
+        bookIt25056em: state.bookItem.filter((book) => book.id !== bookId),
+      };
+    },
+  },
+});
```

Comment:
- [OPTIONAL] Your work is great. I only provide this as an alternative and great way too.

```
removeBook: (state, action) => {
      state.books = state.books.filter((book) => book.id !== action.id)
    },
```